File: shared/src/types/player.ts
import type { CardString } from './card.js';
import type { ActionType } from './hand.js';

export type PlayerStatus =
  | 'sitting_out'
  | 'waiting'
  | 'active'
  | 'folded'
  | 'all_in'
  | 'busted';

export interface Player {
  id: string;
  name: string;
  seatIndex: number;
  stack: number;
  status: PlayerStatus;
  isConnected: boolean;
  isReady: boolean;
  runItTwicePreference: 'ask' | 'always_no';
  autoMuck: boolean;
  disconnectedAt: number | null;
  avatarId: string;
}

export interface PublicPlayerState {
  id: string;
  name: string;
  seatIndex: number;
  stack: number;
  status: PlayerStatus;
  isConnected: boolean;
  disconnectedAt: number | null;
  currentBet: number;
  isDealer: boolean;
  isSmallBlind: boolean;
  isBigBlind: boolean;
  isCurrentActor: boolean;
  holeCards: CardString[] | null;
  hasCards: boolean;
  avatarId: string;
}

export interface PrivatePlayerState {
  id: string;
  name: string;
  seatIndex: number;
  stack: number;
  status: PlayerStatus;
  holeCards: CardString[];
  currentBet: number;
  availableActions: ActionType[];
  minRaise: number;
  maxRaise: number;
  callAmount: number;
  potTotal: number;
  isMyTurn: boolean;
  showCardsOption: boolean;
  runItTwiceOffer: boolean;
  runItTwiceDeadline: number;
}

File: ios/PokerSofta/Sources/Models/Player.swift
import Foundation

enum PlayerStatus: String, Codable {
    case sittingOut = "sitting_out"
    case waiting
    case active
    case folded
    case allIn = "all_in"
    case busted
}

struct PublicPlayerState: Codable {
    let id: String
    let name: String
    let seatIndex: Int
    let stack: Double
    let status: PlayerStatus
    let isConnected: Bool
    let disconnectedAt: Double?
    let currentBet: Double
    let isDealer: Bool
    let isSmallBlind: Bool
    let isBigBlind: Bool
    let isCurrentActor: Bool
    let holeCards: [CardString]?
    let hasCards: Bool
    let avatarId: String
}

struct PrivatePlayerState: Codable {
    let id: String
    let name: String
    let seatIndex: Int
    let stack: Double
    let status: PlayerStatus
    let holeCards: [CardString]
    let currentBet: Double
    let availableActions: [ActionType]
    let minRaise: Double
    let maxRaise: Double
    let callAmount: Double
    let potTotal: Double
    let isMyTurn: Bool
    let showCardsOption: Bool
    let runItTwiceOffer: Bool
    let runItTwiceDeadline: Double
    let sitOutNextHand: Bool
    let autoMuck: Bool
}
